tests/system_tests_http2.py

```diff
@@ -255,7 +255,7 @@ def check_listener_delete(self, client_addr, server_addr):
         # Add back the listener and run a curl command to make sure that the newly added listener is
         # back up and running.
         create_result = qd_manager.create("org.apache.qpid.dispatch.httpListener", self.http_listener_props)
-        sleep(2)
+        sleep(1)
```

Comment:
I really don't like magic timeouts. What about something like this. Some retry method, not sure about precise name, which retries the assertion until either time runs out (then it raises the last error) or until assertion eventually succeeds, in which case it will just move on

```python
def complex_assertion():
    out = self.run_curl(client_addr)
    self.assertIn(ret_string, out)

system_tests.repeat_until(timeout=10, wait=1, f = test(), tolerated_exception=AssertionError)
```